modules/exoplanets/rendering/renderableorbitdisc.cpp

```diff
@@ -0,0 +1,325 @@
+    _uniformCache.eccentricity = _shader->uniformLocation("eccentricity");
+    _uniformCache.semiMajorAxis = _shader->uniformLocation("semiMajorAxis");
+
+    glGenVertexArrays(1, &_quad);
+    glGenBuffers(1, &_vertexPositionBuffer);
+
+    createPlane();
+    loadTexture();
+}
+
+void RenderableOrbitDisc::deinitializeGL() {
+    glDeleteVertexArrays(1, &_quad);
+    _quad = 0;
+
+    glDeleteBuffers(1, &_vertexPositionBuffer);
+    _vertexPositionBuffer = 0;
+
+    _textureFile = nullptr;
+    _texture = nullptr;
+
+    global::renderEngine.removeRenderProgram(_shader.get());
+    _shader = nullptr;
+}
+
+void RenderableOrbitDisc::render(const RenderData& data, RendererTasks&) {
+    _shader->activate();
+
+    glm::dmat4 modelTransform =
+        glm::translate(glm::dmat4(1.0), data.modelTransform.translation) *
+        glm::dmat4(data.modelTransform.rotation) *
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() * modelTransform;
+
+    _shader->setUniform(
+        _uniformCache.modelViewProjection,
+        data.camera.projectionMatrix() * glm::mat4(modelViewTransform)
+    );
+    _shader->setUniform(_uniformCache.textureOffset, _offset);
+    _shader->setUniform(_uniformCache.opacity, _opacity);
+    _shader->setUniform(_uniformCache.eccentricity, _eccentricity);
+    _shader->setUniform(_uniformCache.semiMajorAxis, _size);
+
+    ghoul::opengl::TextureUnit unit;
+    unit.activate();
+    _texture->bind();
+    _shader->setUniform(_uniformCache.texture, unit);
+
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glDepthMask(false);
+    glDisable(GL_CULL_FACE);
+
+    glBindVertexArray(_quad);
+    glDrawArrays(GL_TRIANGLES, 0, 6);
+
+    _shader->deactivate();
+
+    // Restores GL State
+    global::renderEngine.openglStateCache().resetBlendState();
+    global::renderEngine.openglStateCache().resetDepthState();
+    global::renderEngine.openglStateCache().resetPolygonAndClippingState();
+}
+
+void RenderableOrbitDisc::update(const UpdateData& data) {
+    if (_shader->isDirty()) {
+        _shader->rebuildFromFile();
+        _uniformCache.modelViewProjection = _shader->uniformLocation(
+            "modelViewProjectionTransform"
+        );
+        _uniformCache.textureOffset = _shader->uniformLocation("textureOffset");
+        _uniformCache.opacity = _shader->uniformLocation("opacity");
+        _uniformCache.texture = _shader->uniformLocation("discTexture");
+        _uniformCache.eccentricity = _shader->uniformLocation("eccentricity");
+        _uniformCache.semiMajorAxis = _shader->uniformLocation("semiMajorAxis");
+    }
+
+    if (_planeIsDirty) {
+        createPlane();
+        _planeIsDirty = false;
+    }
+
+    if (_textureIsDirty) {
+        loadTexture();
+        _textureIsDirty = false;
+    }
+}
+
+void RenderableOrbitDisc::loadTexture() {
+    if (_texturePath.value() != "") {
```

Comment:
`!_texturePath.value().empty()` would not have to create a temporary empty std::string for the comparison